game/src/contrib/systems/HealthSystem.java

```diff
@@ -108,7 +136,7 @@ private void doDamageAndAnimation(HSData hsd, int dmgAmount) {
     private void removeDeadEntities(HSData hsd) {
         // Entity appears to be dead, so let's clean up the mess
         hsd.hc.triggerOnDeath();
-        hsd.ac.currentAnimation(AdditionalAnimations.DIE);
+        hsd.dc.currentAnimation(AdditionalAnimations.DIE);
```

Comment:
ist das an dieser stelle nicht egal/ zu spät?
An dieser stelle ist das Ding doch tot und die Animation beendet, warum nochmal setzen?

Reply:
vermutlich ein überbleibsel von früher. 

müsste vermutlich (wenn) andersherum gemacht werden, oder? erst sicherheitshalber setzen und dann `hc.triggerOnDeath` aufrufen? dort könnte die sterbe-animation vielleicht noch gebraucht werden. oder es kann hier komplett raus, wenn es denn tatsächlich irgendwo anders gesetzt wird.

Reply:
Wenn ich

```java
.map(this::activateDeathAnimation)
                // Filter by state of animation
                .filter(this::testDeathAnimationStatus)
                // Remove all dead entities
                .forEach(this::removeDeadEntities);
```

richtig lese, dann wird die death Animation gesetzt und wenn `testDeathAnimationStatus` true zurückgibt dann wird das Ding endgültig in `removeDeadEntities` gekillt.

Sollte meines Verständnis nach hier also komplett rausfliegen

Reply:
> dann wird die death Animation gesetzt und wenn testDeathAnimationStatus true zurückgibt dann wird das Ding endgültig in removeDeadEntities gekillt.

so würde ich das auch lesen, ja.

Reply:
ziehe ich doch direkt mal mein approval wieder zurück :)

Reply:
jap, habens gerade im meeting besprochen, ist überflüssig